pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/ingress_test.go

```diff
@@ -0,0 +1,301 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build orchestrator
+// +build orchestrator
+
+package k8s
+
+import (
+	"testing"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestExtractIngress(t *testing.T) {
+	pathType := netv1.PathTypeImplementationSpecific
+	apiGroup := "apiGroup"
+	ingressClassName := "ingressClassName"
```

Comment:
tip 💡: you can use this test helper to build string pointers directly https://github.com/DataDog/datadog-agent/blob/96fea554086547328f93c41d63746aa20ce461ec/pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/common_test.go#L23-L25